tests/unit/amo/containers/TestHome.js

```diff
@@ -104,8 +134,57 @@ describe('Home', () => {
   });
 
   it('maps clientApp to props from state', () => {
-    const { state } = dispatchSignInActions({ clientApp: CLIENT_APP_ANDROID });
+    const { state } = dispatchClientMetadata({ clientApp: CLIENT_APP_ANDROID });
 
     expect(mapStateToProps(state).clientApp).toEqual(CLIENT_APP_ANDROID);
   });
+
+  it('renders a popular extensions shelf', () => {
+    const root = render();
+
+    const shelf = root.find(LandingAddonsCard);
+    expect(shelf).toHaveLength(1);
+    expect(shelf).toHaveProp('header', 'Most popular extensions');
+    expect(shelf).toHaveProp('footerText', 'More popular extensions');
+    expect(shelf).toHaveProp('footerLink', {
+      pathname: '/search/',
+      query: {
+        addonType: ADDON_TYPE_EXTENSION,
+        sort: SEARCH_SORT_POPULAR,
+      },
+    });
+    expect(shelf).toHaveProp('loading', true);
+  });
+
+  it('dispatches an action to fetch the add-ons to display', () => {
```

Comment:
Should we have another test to ensure it renders the content and doesn't set loading once the add-ons have been dispatched?

Reply:
indeed, added new assertions for that.